src/app/learnocaml_teacher_tab.ml

```diff
@@ -299,13 +299,11 @@ let rec teacher_tab token _select _params () =
               H.td [stars_div meta.Exercise.Meta.stars];
               H.td [
                 let cls, text =
-                  match Token.Map.is_empty ES.(st.assignments.token_map),
-                        ES.(st.assignments.default) with
-                  | true, ES.Open -> "exo_open", [%i"Open"]
-                  | true, ES.Closed -> "exo_closed", [%i"Closed"]
-                  | _, (ES.Assigned _ | ES.Closed) ->
-                      "exo_assigned", [%i"Assigned"]
-                  | false, ES.Open -> "exo_assigned", [%i"Open/Assg"]
+                  match ES.(is_open_assg_globally st.assignments) with
+                  | `Open -> "exo_open", [%i"Open"]
```

Comment:
Why did you turn the result type into a polymorphic variant? (Polymorphic variants make sense for extensible code but they provide less guarantee to check for exhaustiveness...)

Reply:
> Why did you turn the result type into a polymorphic variant? (Polymorphic variants make sense for extensible code but they provide less guarantee to check for exhaustiveness...)

Good point, it may be better to use a regular sum type; unfortunately we should use other constructors than `Open`, `Closed` to avoid any clash; maybe `GloballyOpen, GloballyClosed, GloballyOpenOrAssigned, GloballyClosedOrAssigned`